### PR TITLE
Migrate TestCollectionSort.js

### LINKED ISSUE #11181: Migrate /components/TestCollectionControls.js to react-testing-library
Migrate /components/TestCollectionControls.js to react-testing-library

### LINKED ISSUE #11185: Migrate /components/TestCollectionSort.js to react-testing-library
Migrate /components/TestCollectionSort.js to react-testing-library

### PR DESCRIPTION
**Pull request by bobsilverberg**
~~Depends on #11402~~
Fixes #11185 
Fixes #11181

### CODE REVIEW

**Comment by bobsilverberg** [comment]
Note that #11402 should be reviewed first, after which I will rebase this and it will become a fairly small patch.

File: tests/unit/amo/pages/TestCollection.js [hunk 1]
```diff
@@ -1091,4 +1091,30 @@ describe(__filename, () => {
       expect(extractId(props)).toEqual('123/collection-bar/124');
     });
   });
+
+  describe('Tests for CollectionSort', () => {
+    it.each([true, false])(
+      `calls history.push with expected pathname and query when a sort is selected and editing is %s`,
+      (editing) => {
+        const location = `${defaultLocation}${editing ? 'edit/' : ''}`;
```
Comment (eviljeff) [inline tests/unit/amo/pages/TestCollection.js:1099]: this seems to be manually replicating the logic with `editing` in `getLocation`?

**Review by eviljeff — APPROVED** [review]
r+wc

File: tests/unit/amo/pages/TestCollection.js (hunk 1 above)
Comment (bobsilverberg) [reply tests/unit/amo/pages/TestCollection.js:1099]: Good catch. I refactored the test suite a number of times and didn't update this one. I'll do that.

**Pull request merged**